src/main/java/org/traccar/protocol/NavtelecomProtocolDecoder.java

```diff
@@ -221,6 +228,123 @@ protected Object decode(
+                                    position.set("rs485Fuel" + 4, (value < 65500) ? value : null);
+                                    break;
+                                case 42:
+                                    value = buf.readUnsignedShortLE();
+                                    position.set("rs485Fuel" + 5, (value < 65500) ? value : null);
+                                    break;
+                                case 43:
+                                    value = buf.readUnsignedShortLE();
+                                    position.set("rs485Fuel" + 6, (value < 65500) ? value : null);
+                                    break;
+                                case 44:
+                                    value = buf.readUnsignedShortLE();
+                                    position.set("rs232Fuel", (value < 65500) ? value : null);
+                                    break;
+                                case 45:
+                                    value = buf.readByte();
+                                    position.set(Position.PREFIX_TEMP + 1, (value != 0x80) ? value : null);
+                                    break;
+                                case 46:
+                                    value = buf.readByte();
+                                    position.set(Position.PREFIX_TEMP + 2, (value != 0x80) ? value : null);
+                                    break;
+                                case 47:
+                                    value = buf.readByte();
+                                    position.set(Position.PREFIX_TEMP + 3, (value != 0x80) ? value : null);
+                                    break;
+                                case 48:
+                                    value = buf.readByte();
+                                    position.set(Position.PREFIX_TEMP + 4, (value != 0x80) ? value : null);
+                                    break;
+                                case 49:
+                                    value = buf.readByte();
+                                    position.set(Position.PREFIX_TEMP + 5, (value != 0x80) ? value : null);
+                                    break;
+                                case 50:
+                                    value = buf.readByte();
+                                    position.set(Position.PREFIX_TEMP + 6, (value != 0x80) ? value : null);
+                                    break;
+                                case 51:
+                                    value = buf.readByte();
+                                    position.set(Position.PREFIX_TEMP + 7, (value != 0x80) ? value : null);
+                                    break;
+                                case 52:
+                                    value = buf.readByte();
+                                    position.set(Position.PREFIX_TEMP + 8, (value != 0x80) ? value : null);
+                                    break;
```

Comment:
I just realized something. These can be collapsed into something like this:
```suggestion
                                case 45:
                                ...
                                case 52:
                                    value = buf.readByte();
                                    position.set(Position.PREFIX_TEMP + (j - 45), (value != 0x80) ? value : null);
                                    break;
```
And same with fuel.

Reply:
Yes, it looks much more compact. But we need to correct the calculation of the key index. When we get into `case 45`, then `j = 44` and the key index will be `-1`. In order for everything to be calculated correctly, we need to do something like this `(j + 2 - 45)`.